src/ucp/core/ucp_am.c

```diff
@@ -968,9 +960,9 @@ UCS_PROFILE_FUNC(ucs_status_ptr_t, ucp_am_send_nbx,
 
     UCP_WORKER_THREAD_CS_ENTER_CONDITIONAL(worker);
 
-    if (ENABLE_PARAMS_CHECK &&
-        !ucp_am_send_nbx_check_header_length(worker, header_length)) {
-        ret = UCS_STATUS_PTR(UCS_ERR_INVALID_PARAM);
+    status = ucp_am_send_nbx_check_header_length(worker, header_length);
```

Comment:
maybe no need for a separate check function now? Seems to be a simple check and you will not need to check return status here